### ISSUE TITLE
Problem in running the examples using the seamless mode of MATLAB interface

### ISSUE THREAD

**Issue by Pugazhenthi-Sivasankar** [Tudat/tudat#354]
Hello everyone!!! A couple of days ago, I was provided help in fixing the problem in running the unit tests using MATLAB. When I tried to run the example 'singlePerturbedSatellite.m' using MATLAB in seamless mode, I got a similar error: " This application has requested...out_of_range". I tried running the 'json_interface.exe' from the Windows explorer which resulted in a brief display of the command terminal followed by an error message. I have included both the errors in the attachments below:
[SinglePerturbedSatellite_error.txt](https://github.com/Tudat/tudat/files/1819369/SinglePerturbedSatellite_error.txt)
![json_interface](https://user-images.githubusercontent.com/25662615/37523850-744297c6-2928-11e8-8c97-22c95dd530fb.jpg)

**Comment by Pugazhenthi-Sivasankar** [comment]
Dear Dominic,
                     There was no problem when I ran the file 'singlePerturbedSatellite.m' that I downloaded from the GitHub tudatBundle repository. However, when I made changes to the file according to my requirements, I could not run it. I commented the line 'deleteIfExists(Simulation.defaultInputFileName);' from the 'Simulation.m' which resulted in the creation of the file 'temp.json'. Since I did not know how to pass 'temp.json' as an input argument to 'json_interface.exe', I searched the internet for solutions. In one website, it was mentioned to make use of 'jq' package to convert JSON files into usable format within the command prompt. When I downloaded the 'jq' package, my antivirus discarded it as it was a threat to my system. I do not know how to run the 'json_interface.exe' from the Qt creator. I need your advice in this situation.

**Comment by DominicDirkx** [comment]
http://tudat.tudelft.nl/tutorials/jsonInterface/basicJsonInterfaces.html

**Comment by Pugazhenthi-Sivasankar** [comment]
I followed the procedure mentioned in the website and got the following error messages. The first is when I tried to run it from the terminal and the second is when I tried to run it from Qt creator.
![json_interface_temp_json](https://user-images.githubusercontent.com/25662615/37706963-9c8e3316-2d01-11e8-88ba-7cf0325fa5c9.jpg)
[json_inter_temp_json_Qt.txt](https://github.com/Tudat/tudat/files/1832983/json_inter_temp_json_Qt.txt)

**Comment by DominicDirkx** [comment]
Hi Pugazh,

Under 'Run Settings', do you have 'Run in Terminal' enabled? If so, disable it, and run again,

Dominic

**Comment by Pugazhenthi-Sivasankar** [comment]
Hello Dominic,
                    I ran other applications in TUDAT and it went well without any errors. Out of the 213 unit tests, 3 failed and I have posted the output in a text file below:
[unit_test_error.txt](https://github.com/Tudat/tudat/files/1833578/unit_test_error.txt)

**Comment by DominicDirkx** [comment]
Hi Pugazh,

Just to make sure, post the test output files here (see Tudat website), but I don't think these failures are related to your problem. Could you post a screenshot of your Run Settings?

**Comment by Pugazhenthi-Sivasankar** [comment]
Here is a screenshot of the Build settings used while running the unit tests:
![build_settings](https://user-images.githubusercontent.com/25662615/37791946-bf1d62bc-2e0a-11e8-9b07-8d5f0966c007.jpg)
Here is a screenshot of the Run settings used while running the unit tests:
![run_settings](https://user-images.githubusercontent.com/25662615/37791996-dce23764-2e0a-11e8-9d1c-9fa8e4273ef3.jpg)

**Comment by DominicDirkx** [comment]
Hi Pugazh,

Since the unit tests seem to be doing ok, no need to post anything on them, please post your build and run settings for the test you are running - which seems to be failing. Just so you know, I probably won't have time to look at this before Monday,

Dominic

**Comment by Pugazhenthi-Sivasankar** [comment]
Here is a screenshot of the Build settings used while running the 'json_interface.exe' file:
![build_json_interface](https://user-images.githubusercontent.com/25662615/37826742-fe0c8608-2e94-11e8-8ed3-775d445d3fa0.jpg)
Here is a screenshot of the Run settings used while running the 'json_interface.exe' file:
![run_json_interface](https://user-images.githubusercontent.com/25662615/37826782-25b9d0b6-2e95-11e8-8e61-d4b4d138791d.jpg)

**Comment by DominicDirkx** [comment]
You must specify the input argument, which is your input file (temp.json). 

If this also does not work: does the json_interface perform well when you provide it with an input file from one of the examples? If yes, there is something wrong with the input json file you're using. 

You can also run a json file in an alternative way, bu using the JsonSimulationManager class, see for example the code around line 50 in:

/tudat/Tudat/JsonInterface/UnitTests/unitTestSimulationGalileoConstellation.cpp

You provide the JSON file as input argument to the JsonSimulationManager constructor. Since these unit tests passed for you, this interface will work. If it doesn't work with your input file, the issue is definitely there.

**Comment by Pugazhenthi-Sivasankar** [comment]
First, I ran the code 'unitTestSimulationGalileoConstellation.cpp' and it produced the expected output. Second, I used the JsonSimulationManager class in a separately created source file called 'json_testing.cpp' and input the json file from one of the examples given in the seamless folder. But it showed errors during compilation. These are given in this file: 
[json_default.txt](https://github.com/Tudat/tudat/files/1847608/json_default.txt)
Third, I input the json file corresponding to one of the satellites I am working with and this also produced errors during compilation. These are given in this file:
[json_PROBA_1.txt](https://github.com/Tudat/tudat/files/1847614/json_PROBA_1.txt)

**Comment by DominicDirkx** [comment]
The JSON file you use won't influence compilation errors, these are only parsed during runtime. If you push your code to Github and post a link here, I can have a look where these compilation errors come from.

**Comment by Pugazhenthi-Sivasankar** [comment]
I have added the files 'json_testing.cpp', 'main.json' and 'temp.json' in the repository Pugazh1. Here is the link: https://github.com/Pugazhenthi-Sivasankar/Pugazh1

**Comment by DominicDirkx** [comment]
Replace "main.json" with std::string( "main.json" ), and it should compile.

**Comment by Pugazhenthi-Sivasankar** [comment]
After replacing the required content, I tried compiling. It resulted in 1095 issues. I have included them in the file here:
[json_issues_1095.txt](https://github.com/Tudat/tudat/files/1848510/json_issues_1095.txt)

**Comment by DominicDirkx** [comment]
This looks like your target_link_libraries defined for this executable in the CMakeLists.txt file are missing something. Look at the JSON  executables in th example application/unit test CMakeLists.txt, and make sure you link against the same libraries.

**Comment by Pugazhenthi-Sivasankar** [comment]
This time I got 421 issues while compiling. I have included them in the file here: 
[json_issues_421.txt](https://github.com/Tudat/tudat/files/1848845/json_issues_421.txt)
For your reference, I have included the corresponding 'CMakeLists.txt' file in the link:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1

**Comment by DominicDirkx** [comment]
Hi Pugazh,

For now, try using the existing JSON applications/unit tests to run your JSON file. They cpmpile, and you can change the input parameter to your filename. At the earliest, I will have time on Thursday to look at this in more detail,

Dominic

**Comment by Pugazhenthi-Sivasankar** [comment]
I ran the default 'unitTestSimulationSinglePerturbedSatellite.cpp' in the Qt creator and it worked well. Next, I replaced the contents of this file with the contents of 'json_testing.cpp' and used the file 'temp.json' (this was created while running the seamless example 'singlePerturbedSatellite.m'). Though there were some issues, I was able to obtain the specified output. Next, I used the file 'main.json' (this was created while running the customised seamless code 'singlePerturbedSatellite_PROBA_1.m') in the place of 'temp.json' and ran the program. This resulted in compilation errors and I could not get any output. The errors are included in the file: 
[main_json_compilation_errors.txt](https://github.com/Tudat/tudat/files/1853334/main_json_compilation_errors.txt)
For your reference, I have included the file 'unitTestSimulationSinglePerturbedSatellite.cpp' in the following link: https://github.com/Pugazhenthi-Sivasankar/Pugazh1

**Comment by DominicDirkx** [comment]
Your compile output seems to be succesful, just a few warnings, it ends in succesful compilation of the test_JsonInterfaceSimulationSinglePerturbedSatellite. 

There's plenty you can try yourself to pin down where it's going wrong, if anything is going wrong. For instance, put the contents of one of your JSON files into one of the example JSON files, and use that. If this does not run succesfully, you can be sure that it was your JSON file that's the problem. If it does, it's the way you're calling it...

**Comment by Pugazhenthi-Sivasankar** [comment]
I ran 'unitTestSimulationSinglePerturbedSatellite.cpp' from the Qt creator and 'test_JsonInterfaceSimulationSinglePerturbedSatellite.exe' from the command prompt, both using the 'temp.json' as input. Both of them gave the expected results. When I tried the same using 'main.json' as input, the following errors showed up:
[main_json_application_output.txt](https://github.com/Tudat/tudat/files/1863259/main_json_application_output.txt)
[main_json_terminal_output.txt](https://github.com/Tudat/tudat/files/1863260/main_json_terminal_output.txt)
When I compared the files 'main.json' and 'temp.json', there seems to be no difference than the names of the satellites and the respective initial conditions.

**Comment by Pugazhenthi-Sivasankar** [comment]
For now, I got the problem solved. I had two versions of MATLAB in my laptop (R2014b and R2016b). I uninstalled the old version and made changes to the MATLAB script for PROBA_1. There was a line 'proba_1 = Body('PROBA_1');' which created the problem. I changed it to 'PROBA_1 = Body('PROBA_1');' and used PROBA_1 elsewhere in the document. This resulted in the expected output of the program.
